### PR TITLE
docs: add gsoc 2022 work report

### PR DESCRIPTION
**Pull request by rajatgupta24**
# Description
Writing a blog post about the gsoc 2022 work report.

# Checklist:

- [x] I have mentioned the appropriate type(scope), as referenced [here](https://jenkins-x.io/community/code/#the-commit-message) in the commit message and PR title for the semantic checks to pass.
- [x] I have signed off the commit, as per instructions mentioned [here](https://jenkins-x.io/community/code/#how-to-sign-your-commits).
- [x] Any dependent changes have already been merged.

### CODE REVIEW

**Comment by rajatgupta24** [comment]
The exact link to the blog preview: 
https://jx-docs-jx-jenkins-x-jx-docs-pr-3673.infra.jenkins-x.rocks/blog/2022/11/13/gsoc-2022-work-report/

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+- RBAC to prevent the distribution of cluster access.
+
+New Jenkins X UI focus on Simplicity, Security and a Superb User Experience.
+
+This is NOT GA(General Availability) yet.
+
+Visit the project repo [here](https://github.com/jenkins-x/jx-ui) to try it.
+
+## Work Done
+### Stop a running or pending PipelineActivity
```
Comment (babadofar) [inline content/en/blog/news/2022-gsoc-work-report.md:34]: Remove this section? Is this really under what has bben done? It shows the "jx" command

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+**Issue:** https://github.com/jenkins-x/jx-ui/issues/28
+
+**PRs:**
+- https://github.com/jenkins-x/jx-ui/pull/50
+- https://github.com/jenkins-x/jx-ui/pull/51
+- https://github.com/jenkins-x/jx-ui/pull/59
+- https://github.com/jenkins-x/jx-ui/pull/75
+
+
+### Show message with status of the PipelineActivity
```
Comment (babadofar) [inline content/en/blog/news/2022-gsoc-work-report.md:52]: Is this done? I not remove it, and describe more about what you did. Write something about Svelte and how you have a golong backend and a Svelte fronted serverd in the same container

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+When a pipeline is cancelled or it has timedout, we update the status of the pipeline to cancelled or timedout. But we dont show any reason why a job was timedout. We should show that information in the pipeline details page (along with the step which has timed out).
+
+**Issue:** https://github.com/jenkins-x/jx-ui/issues/23
+
+**PRs:**
+- https://github.com/jenkins-x/jx-api/pull/171
+- https://github.com/jenkins-x-plugins/jx-pipeline/pull/481
+
+
+### Implement a DAG for PipelineActivity
```
Comment (babadofar) [inline content/en/blog/news/2022-gsoc-work-report.md:62]: Add images to show what it looks like

**Review by babadofar — CHANGES_REQUESTED** [review]

File: content/en/blog/news/2022-gsoc-work-report.md (hunk 2 above)
Comment (rajatgupta24) [reply content/en/blog/news/2022-gsoc-work-report.md:52]: It is done in the backend but still needs to be added in the frontend.

> Write something about Svelte and how you have a golong backend and a Svelte fronted serverd in the same container

I was thinking of adding that in the intro section.

File: content/en/blog/news/2022-gsoc-work-report.md (hunk 1 above)
Comment (rajatgupta24) [reply content/en/blog/news/2022-gsoc-work-report.md:34]: Yes, this is done, we can now stop a pipeline from UI, I'll rewrite the explanation.

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+description: This project replaces the older UI with the new UI with more powerful features.
+categories: [blog]
+keywords: [Community, jenkins-x, gsoc, 2022]
+slug: "GSoC-2022-work-report"
+aliases: []
+author: Rajat Gupta
+---
+
+## Jenkins X New UI
+It is a helm chart that you can install with the other tools of jx. It is build using [Golang](https://go.dev/) as backend and [Sveltekit](https://kit.svelte.dev/) as frontend, both backend and frontend are build together and are used in the same container.
```
Comment (babadofar) [inline content/en/blog/news/2022-gsoc-work-report.md:14]: Start saying: "It is a web application built with Golang ... (next sentence). Add info on helm chart next. ("It is available as a helm chart so you can install like any other Kubernetes component") ("But needs Jenkins X CRDs to work properly?!)

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+- Add that domain to the ingress file, Run the following command and replace **dashboard.jx.change.me** with ngrok link.
+  ```
+  kubectl edit ing jx-ui
+  ```
+- And boom, visit localhost:9200 and it works, you should see the homepage screen.
+
+## Work Done
+### Stop a running or pending PipelineActivity from UI
+
+The only way to stop a running or pending PipelineActivity is to run `jx pipeline stop`, which is a cli command, the old UI does not provide this feature.
```
Comment (babadofar) [inline content/en/blog/news/2022-gsoc-work-report.md:60]: Remove this sentence

Comment (babadofar) [reply content/en/blog/news/2022-gsoc-work-report.md:60]: Or rewrite it and add it after the one below.

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+![pipelinesPage.png](/images/gsoc-work-report-22/pipelinesPage.png)
+
+## Why need a new UI?
+A good UI is essential for a CI/CD tool, as not everyone is familiar with the CLI. The current UI (jx-pipeline-visualizer) is a read-only UI, the user can view the logs of PipelineActivity but neither can start nor stop the pipeline.
+
+Features that UI will provide: 
+- Start and Stop a PipelineActivity.
+- Have a audit trail.
+- A DAG representation of PipelineActivity.
```
Comment (babadofar) [inline content/en/blog/news/2022-gsoc-work-report.md:28]: Replace DAG with "graphical representation" ? It is more general, that it is a DAG is an implementation detail (although important)

Comment (rajatgupta24) [reply content/en/blog/news/2022-gsoc-work-report.md]: **DAG** is a popular term but **graphical representation** is the more general term.

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+description: This project replaces the older UI with the new UI with more powerful features.
+categories: [blog]
+keywords: [Community, jenkins-x, gsoc, 2022]
+slug: "GSoC-2022-work-report"
+aliases: []
+author: Rajat Gupta
+---
+
+## Jenkins X New UI
+It is a web application built with [Golang](https://go.dev/) for the backend and [Sveltekit](https://kit.svelte.dev/) for the frontend, both of which are built together and used in the same container. To function properly, it must be installed as a helm chart with Jenkins X CRDs.
```
Comment (ankitm123) [inline content/en/blog/news/2022-gsoc-work-report.md:14]: Would be great to follow this: https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+## What's next?
+
+- The UI project is just started, we have also start a seperate **UI-sig** which you can join from [calendar](https://jenkins-x.io/community/#office-hours).
+- There are some amazing features that we are planning to add to the new UI: 
+  - Dynamically update the list of pipelines
+  - Upgrade version stream from the UI
+  - Import repositories using the UI
+  - Break down pipeline logs into individual steps
+  - Mask secrets in the logs
+  - Store logs for cancelled and timed out pipelines
```
Comment (ankitm123) [inline content/en/blog/news/2022-gsoc-work-report.md:115]: Not directly related to the UI, as that needs to be implemented in the build controller ( or jx-pipeline codebase). So may be change this to `display logs for cancelled and timedout pipelines`

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+**PRs:**
+- https://github.com/jenkins-x/jx-ui/pull/50
+- https://github.com/jenkins-x/jx-ui/pull/51
+- https://github.com/jenkins-x/jx-ui/pull/59
+- https://github.com/jenkins-x/jx-ui/pull/75
+
+
+### Show message with status of the PipelineActivity
+
+When a pipeline is cancelled or it has timedout, we update the status of the pipeline to cancelled or timedout.
```
Comment (ankitm123) [inline content/en/blog/news/2022-gsoc-work-report.md:85]: `timed out` (separate words)

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+But we dont show any reason why a job was timedout. We should show that information in the pipeline details page (along with the step which has timed out).
```
Comment (ankitm123) [inline content/en/blog/news/2022-gsoc-work-report.md:86]: timed out (separate words)

Also separate lines.

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+**Issue:** https://github.com/jenkins-x/jx-ui/issues/23
+
+**PRs:**
+- https://github.com/jenkins-x/jx-api/pull/171
+- https://github.com/jenkins-x-plugins/jx-pipeline/pull/481
+
+
+### Implement a DAG for PipelineActivity
+
+To make the UI more user friendly and to explain the sequence of the steps in a PipelineActivity, we have added a DAG (Directed Acyclic Graph) chart to graphically visualize the PipelineActivity.
```
Comment (ankitm123) [inline content/en/blog/news/2022-gsoc-work-report.md:97]: May be explain DAG? or add a link to it (a wiki link is fine)?

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+  - Break down pipeline logs into individual steps
+  - Mask secrets in the logs
+  - Display logs for cancelled and timed out pipelines
+
+To read more read the issues created [here](https://github.com/jenkins-x/jx-ui/issues)
+
+## Acknowledgements
+
+I never imagined I would be this sad after finishing an internship.
+Just as Osama noted in his work report, this was the most difficult experience we have had thus far, but the encouragement of the jx community and the direction of our mentor made it the best educational experience I have ever had.
```
Comment (ankitm123) [inline content/en/blog/news/2022-gsoc-work-report.md:127]: probably link to osama's post, if you are mentioning him?

File: content/en/community/_index.md [hunk 13]
```diff
@@ -63,5 +63,3 @@ Got any thoughts, ideas or feedback then please drop them in a [GitHub Issue](ht
 You may also want to try [Stack Overflow](https://stackoverflow.com/questions/tagged/jenkins--x) with the tag `jenkins--x` though we prefer [GitHub Issues](https://github.com/jenkins-x/jx/issues).
 
 ### Contributors
-
-{{< contributors >}}
```
Comment (ankitm123) [inline content/en/community/_index.md:67]: Why this change?

**Comment by ankitm123** [comment]
```


htmlproofer 3.8.0 \| Error:  HTML-Proofer found 1 failure!
--
  |   |   |   | - ./tmp-website/blog/2022/11/13/gsoc-2022-work-report/index.html
  |   |   |   | *  internally linking to content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md, which does not exist (line 706)
  |   |   |   | <a href="content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md">work report</a>


```

File: content/en/blog/news/2022-gsoc-work-report.md [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+
+Features that the UI will provide: 
+- Start and Stop a PipelineActivity.
+- Have an audit trail.
+- A graphical representation of PipelineActivity.
+- RBAC to limit access to certain functionalities.
+
+New Jenkins X UI focus on Simplicity, Security and a Superb User Experience.
+
+This is NOT GA(General Availability) yet. Visit the project repo [here](https://github.com/jenkins-x/jx-ui) to try it.
```
Comment (ankitm123) [inline content/en/blog/news/2022-gsoc-work-report.md:35]: Needs space between `GA and (`

**Review by ankitm123 — APPROVED** [review]

**Pull request merged**